src/user/dto/create-user.dto.ts

```diff
@@ -1,12 +1,45 @@
-import { Gender, UserRole } from "src/constants";
+import { ApiProperty } from '@nestjs/swagger';
+import { Gender, UserRole } from 'src/constants';
+import { IsNotEmpty, IsEmail, IsPhoneNumber, IsDateString, IsEnum, Matches } from 'class-validator';
 
 export class CreateUserDto {
-  fullname: string;
+  @ApiProperty()
+  @IsNotEmpty({ message: 'Full name must not be empty' })
+  fullName: string;
+
+  @ApiProperty()
+  @IsNotEmpty({ message: 'Username must not be empty' })
+  @Matches(/^\S*$/, { message: 'Username must not contain spaces' })
```

Comment:
I think you don't need to use @Matches validator here, you can you @Transform to transform data before validate.
Refer:
https://github.com/typestack/class-transformer
https://stackoverflow.com/questions/55333276/validating-empty-space-in-class-validator.

Suggestion:
```
@IsNotEmpty()
@Transform(({ value }: TransformFnParams) => value?.trim())
userName: string;
```
Please apply for other properties.